From 7ca657d2f9fe95bea0f44599db3bc752c7faa0b5 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 12 Aug 2021 17:31:29 +0300
Subject: [PATCH] [SW-0] gaudi: security: improve lbw handover fix

HOST_PHYS_BASE does not fit the 32bit register and therefore there will
be 0 written to it and then extended to 0x7F00000000.
This is not what we meant and will not work as expected.
Subtract 1 from HOST_PHYS_BASE to hae a valid value.

Change-Id: I3b43ed2f4af23785e6c1bace3864e88845a1d2d9
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index c3d21b5..5911e25 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1501,6 +1501,7 @@
  */
 #define CFG_HIGH_GAP_BASE	0x7FFCFC9000ull
 #define DBG_BASE		mmMME_S_ROM_TABLE_BASE
+#define HOST_PHYS_BASE		0x8000000000ull
 
 static void preboot_secure_lbw(void)
 {
@@ -1524,7 +1525,14 @@
 
 	/* CFG HIGH RANGE: RR(PCIE_DYN_RR_CFG_HIGH) */
 	min = mmPSOC_GPIO0_BASE - 1;
-	max = HOST_PHYS_BASE;
+	/*
+	 * The PCIe RR can take only values of 32 bits, then H/W is
+	 * extending these with 0x7F in the bits 32-39.
+	 * This means we cannot provide the HOST_PHYS_BASE as it will
+	 * just turn into 0x7F00000000 which is not what we want.
+	 * Subtract 1 from HOST_PHYS_BASE to hae a valid value.
+	 */
+	max = HOST_PHYS_BASE - 1;
 	preboot_secure_pcie_rr(PCIE_DYN_RR_CFG_HIGH, min, max);
 
 	/* MME1_QM: RR(LBW_RR_MME1_QM) */
@@ -1897,7 +1905,6 @@
 
 #define SRAM_SIZE	0x1400000	/* 20MB */
 #define SRAM_END	(SDRAM_BASE + SRAM_SIZE)
-#define HOST_PHYS_BASE	0x8000000000ull
 
 enum preboot_sec_rr_names {
 	SRAM_BGN_TO_VERS = 0,
